src/lib/Scenes/MyCollection/Screens/Artwork/Components/MyCollectionArtworkHeader.tsx

```diff
@@ -1,86 +1,57 @@
-import { Size } from "lib/Scenes/Artwork/Components/ImageCarousel/geometry"
-import { MyCollectionImageView } from "lib/Scenes/MyCollection/Components/MyCollectionImageView"
+import {
+  ImageCarousel,
+  ImageCarouselFragmentContainer,
+} from "lib/Scenes/Artwork/Components/ImageCarousel/ImageCarousel"
 import { ScreenMargin } from "lib/Scenes/MyCollection/Components/ScreenMargin"
-import { useDevToggle } from "lib/store/GlobalStore"
 import { retrieveLocalImages } from "lib/utils/LocalImageStore"
 import { useScreenDimensions } from "lib/utils/useScreenDimensions"
-import { Button, Flex, Spacer, Text } from "palette"
-import React from "react"
-import { TouchableOpacity } from "react-native"
-import { createRefetchContainer, graphql, RelayRefetchProp } from "react-relay"
+import { Box, Spacer, Text, useColor } from "palette"
+import React, { useEffect, useState } from "react"
+import { createFragmentContainer, graphql } from "react-relay"
 import { useTracking } from "react-tracking"
-import useInterval from "react-use/lib/useInterval"
-import {
-  getBoundingBox,
-  getImageMeasurements,
-  hasImagesStillProcessing,
-  imageIsProcessing,
-  isImage,
-} from "../../ArtworkForm/MyCollectionImageUtil"
+import { ImageDescriptor } from "../../../../../Scenes/Artwork/Components/ImageCarousel/ImageCarouselContext"
+import { hasImagesStillProcessing } from "../../ArtworkForm/MyCollectionImageUtil"
 
 interface MyCollectionArtworkHeaderProps {
   artwork: MyCollectionArtworkHeader_artwork
-  relay: RelayRefetchProp
 }
 
 export const MyCollectionArtworkHeader: React.FC<MyCollectionArtworkHeaderProps> = (props) => {
   const {
     artwork: { artistNames, date, images, internalID, title, slug },
-    relay,
   } = props
+
+  const [imagesToDisplay, setImagesToDisplay] = useState<typeof images | ImageDescriptor[] | null>(images)
+  const [isDisplayingLocalImages, setIsDisplayingLocalImages] = useState(false)
+
   const dimensions = useScreenDimensions()
   const formattedTitleAndYear = [title, date].filter(Boolean).join(", ")
 
-  const defaultImage = images?.find((i) => i?.isDefault) || (images && images[0])
-
-  const showLocalImages = useDevToggle("DTMyCollectionShowLocalImages")
+  const color = useColor()
 
   const { trackEvent } = useTracking()
 
-  useInterval(() => {
-    if (!isImage(defaultImage) || hasImagesStillProcessing(defaultImage, images)) {
-      relay.refetch(
-        {
-          artworkID: slug,
-        },
-        null,
-        null,
-        { force: true }
-      )
-    }
-  }, 1000)
-
-  const renderMainImageView = () => {
-    const maxImageHeight = dimensions.height / 2.5
-    const imageSize: Size = {
-      height: defaultImage?.height ?? maxImageHeight,
-      width: defaultImage?.width ?? dimensions.width,
+  useEffect(() => {
+    const defaultImage = images?.find((i) => i?.isDefault) || (images && images[0])
+    if (hasImagesStillProcessing(defaultImage, images)) {
```

Comment:
there is a slight problem here with the case where the user edits the artwork and adds an image, we are clearing out local images for simplicity here but the `hasImagesStillProcessing` will return true and there will be no local images so the user is going to see the fallback image even though we have valid processed images.

Think for simplicities sake for the time being we should just check if the main image is still processing and only use local images in that case, but we might want to have more robust logic here eventually, e.g support showing both local and remote images in carousel or showing a placeholder view for processing images in carousel.

```suggestion
   if (!isImage(defaultImage) || imageIsProcessing(defaultImage, "normalized")) {
```

Reply:
Thanks. Super! 👏